### ISSUE TITLE
xgboost need normalization preprocessing?

### ISSUE THREAD

**Issue by luoruisichuan** [dmlc/xgboost#2621]
For decision tree algorithm, normalizaton is not necessary in trainning model procedure.  Xgboost is a kind of decision tree algorithm, but on the other hand,  xgboost use gradient descent method. So my question is that  where  feathures normalization is necessary for xgboost ?
thanks

For bugs or installation issues, please provide the following information.
The more information you provide, the more easily we will be able to offer
help and advice.

## Environment info
Operating System:

Compiler:

Package used (python/R/jvm/C++):

`xgboost` version used:

If installing from source, please provide 

1. The commit hash (`git rev-parse HEAD`)
2. Logs will be helpful (If logs are large, please upload as attachment).

If you are using jvm package, please 

1. add [jvm-packages] in the title to make it quickly be identified
2. the gcc version and distribution

If you are using python package, please provide

1. The python version and distribution
2. The command to install `xgboost` if you are not installing from source

If you are using R package, please provide

1. The R `sessionInfo()`
2. The command to install `xgboost` if you are not installing from source

## Steps to reproduce

1.
2.
3.

## What have you tried?

1.
2.
3.

**Comment by pommedeterresautee** [comment]
XGBoost doesn't use gradient descent, but boosting. Sometimes boosting is compared to gradient descent (https://en.wikipedia.org/wiki/Gradient_boosting).

In XGBoost you can choose to boost a linear model or a tree. Usually trees give better results. If you use a trees, then you don't need to normalize your data.

**Comment by Bingohong** [comment]
tqchen answered the question ,"no need to normalize features" . reference by [Is Normalization necessary?](https://github.com/dmlc/xgboost/issues/357)

**Comment by luoruisichuan** [comment]
thanks. 
Another question, for the case of this condition, objective is 'reg：linear' and eval_metric is 'rmse',  does normalization is needed? @tqchen
thanks very much.

**Comment by coolboygym** [comment]
@Bingohong  Thanks !
